SmartDeviceLink/SDLLockScreenPresenter.m

```diff
@@ -8,61 +8,108 @@
+        [[NSNotificationCenter defaultCenter] postNotificationName:SDLLockScreenManagerWillPresentLockScreenViewController object:nil];
+
+        CGRect firstFrame = appWindow.frame;
+        firstFrame.origin.x = CGRectGetWidth(firstFrame);
+        appWindow.frame = firstFrame;
+
+        // We then move the lockWindow to the original appWindow location.
+        self.lockWindow.frame = appWindow.bounds;
+        [self.lockWindow makeKeyAndVisible];
+
+        // And present the lock screen.
+        SDLLogD(@"Present lock screen window");
+        [self.lockWindow.rootViewController presentViewController:self.lockViewController animated:YES completion:^{
+            // Tell ourselves we are done.
+            [[NSNotificationCenter defaultCenter] postNotificationName:SDLLockScreenManagerDidPresentLockScreenViewController object:nil];
+        }];
     });
 }
 
 - (void)dismiss {
-    if (!self.viewController) {
-        return;
-    }
-
     dispatch_async(dispatch_get_main_queue(), ^{
-        [self.viewController.presentingViewController dismissViewControllerAnimated:YES completion:nil];
+        NSArray *windows = [[UIApplication sharedApplication] windows];
+        UIWindow *appWindow = windows.firstObject;
+
+        if (appWindow.isKeyWindow || appWindow == self.lockWindow) {
+            return;
+        }
+
+        // Let us know we are about to dismiss.
+        [[NSNotificationCenter defaultCenter] postNotificationName:SDLLockScreenManagerWillDismissLockScreenViewController object:nil];
+
+        // Dismiss the lockscreen
+        SDLLogD(@"Dismiss lock screen window");
+        [self.lockViewController dismissViewControllerAnimated:YES completion:^{
+            CGRect lockFrame = self.lockWindow.frame;
+            lockFrame.origin.x = CGRectGetWidth(lockFrame);
+            self.lockWindow.frame = lockFrame;
+
+            // Quickly move the map back, and make it the key window.
+            appWindow.frame = self.lockWindow.bounds;
+            [appWindow makeKeyAndVisible];
+
+            // Tell ourselves we are done.
+            [[NSNotificationCenter defaultCenter] postNotificationName:SDLLockScreenManagerDidDismissLockScreenViewController object:nil];
+        }];
     });
 }
 
 - (BOOL)presented {
-    if (!self.viewController) {
-        return NO;
-    }
-
-    __block BOOL presented = NO;
-    if (![NSThread isMainThread]) {
+    __block BOOL isPresented = NO;
+    if ([NSThread isMainThread]) {
+        isPresented = [self sdl_presented];
+    } else {
         dispatch_sync(dispatch_get_main_queue(), ^{
-            presented = (self.viewController.isViewLoaded && (self.viewController.view.window || self.viewController.isBeingPresented));
+            isPresented = [self sdl_presented];
         });
-    } else {
-        presented = (self.viewController.isViewLoaded && (self.viewController.view.window || self.viewController.isBeingPresented));
     }
-
-    return presented;
+    
+    return self;
```

Comment:
The lock screen is no longer working because the wrong value is being returned for `presented`. If `return self` is changed to `return isPresented` the lock screen works as expected.

Reply:
Fixed